cleanlab/object_detection/summary.py

```diff
@@ -444,6 +449,138 @@ def visualize(
     plt.show()
 
 
+def _get_per_class_confusion_matrix_dict_(
+    labels: List[Dict[str, Any]],
+    predictions: List[np.ndarray],
+    iou_threshold: Optional[float] = 0.5,
+    num_procs: int = 1,
+) -> DefaultDict[int, Dict[str, int]]:
+    """
+    Returns a confusion matrix dictionary for each class containing TP,FP and FN.
+    """
+    num_classes = len(predictions[0])
+    num_images = len(predictions)
+    pool = Pool(num_procs)
+    counter_dict: DefaultDict[int, collections.Counter[int]] = collections.defaultdict(
+        collections.Counter
+    )
+
+    for class_num in range(num_classes):
+        pred_bboxes, lab_bboxes = _filter_by_class(labels, predictions, class_num)
+        tpfp = pool.starmap(
+            _calculate_true_positives_false_positives,
+            zip(
+                pred_bboxes,
+                lab_bboxes,
+                [iou_threshold for _ in range(num_images)],
+                [True for _ in range(num_images)],
+            ),
+        )
+        for j, tpfp_j in enumerate(tpfp):
+            for k, tpfp_k in enumerate(tpfp_j):
+                counter_dict[class_num][k] += np.sum(tpfp_k)
+
+    results: DefaultDict[int, Dict[str, int]] = collections.defaultdict(dict)
+    for class_num in counter_dict:
+        results[class_num]["TP"] = counter_dict[class_num][0]
+        results[class_num]["FP"] = counter_dict[class_num][1]
+        results[class_num]["FN"] = counter_dict[class_num][2]
+    return results
+
+
+def get_average_per_class_confusion_matrix_(
```

Comment:
Naming convention for function is strange, is this function public or private? If it is private it should be `_get_average_per_class_confusion_matrix()` if public then `get_average_per_class_confusion_matrix()`and it should contain a detailed docsting summarizing what it does.

Reply:
I assumed this function is public-facing, so I added the docstring summarization and deleted the trailing `_` in the new commit.